### PR TITLE
refactor(FOROME-954): add async stores for dtree_set & dtree_counts

### PR DESCRIPTION
**Pull request by QYaroslavTrefilov**
https://quantori.atlassian.net/browse/FOROME-954

Implement async stores for dtree_set & dtree_counts request

### CODE REVIEW

File: src/store/dtree.ts [hunk 1]
```diff
@@ -114,7 +114,44 @@ class DtreeStore {
...
+
+    return counts ?? []
+  }
+
   constructor() {
+    reaction(
+      () => this.dtreeSetData,
+      response => {
+        if (response) {
+          stepStore.setSteps(adaptDtreeSetToSteps(response))
```
Comment (AndreyUstyumenko) [inline src/store/dtree.ts]: adapters are not utility functions. They are supposed to modify the outcome data interface. They work like this: Interface A --> adapter --> Interface B. `getDtreeSet` should return another interface, which would include `steps` inside of it. In other words, you must apply this adapter in `getDtreeSet` method and add `steps` to the response. And in this place you're going to call just `this.dtreeSetData?.steps`.

Comment (AndreyUstyumenko) [inline src/store/dtree.ts]: adapters are not utility functions. They are supposed to modify the outcome data interface. They work like this: Interface A --> adapter --> Interface B. `getDtreeSet` should return another interface, which would include `steps` inside of it. In other words, you must apply this adapter in `getDtreeSet` method and add `steps` to the response. And in this place you're going to call just `this.dtreeSetData?.steps`.

**Comment by QYaroslavTrefilov** [comment]
typo in last commit:
"fix(FOROME-1029): make final step active after delete different step"

**Review by AndreyUstyumenko — APPROVED** [review]

File: src/service-providers/decision-trees/decision-trees.adapters.ts [hunk 2]
```diff
@@ -11,3 +18,67 @@ export const adaptDtreeStatResponse = (
...
+      conditionPointIndex,
+      returnPointIndex: conditionPointIndex + 1,
+      comment: stepCodes[index].comment,
+      negate: stepCodes[index].isNegate,
+      all: stepCodes[index].isAll,
+      condition: stepCodes[index].condition,
+    })
+  })
+
+  localSteps.forEach((step: IStepData, index: number) => {
```
Comment (AlMaQntr) [inline src/service-providers/decision-trees/decision-trees.adapters.ts]: @Lapk1n removed this in his PR.
And also, he refactored some fields in `localSteps`

**Review by AlMaQntr — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Pull request merged**